File: app/src/main/java/com/bq/zowi/models/commands/ForwardRunnerCommand.java
package com.bq.zowi.models.commands;

import com.bq.zowi.models.commands.Command;

/* JADX INFO: loaded from: classes.dex */
public class ForwardRunnerCommand extends MovementCommand {
    private static final Command.Action[] FORWARD_ACTION = {Command.Action.WALK};
    private static final Command.Direction[] FORWARD_DIRECTION = {Command.Direction.FORWARD};

    public ForwardRunnerCommand(Command.Action action, Command.Direction direction, long duration) {
        setAction(action);
        setDirection(direction);
        setDuration(duration);
    }

    @Override // com.bq.zowi.models.commands.Command
    public Command.Action[] getAllowedActions() {
        return FORWARD_ACTION;
    }

    @Override // com.bq.zowi.models.commands.Command
    public Command.Direction[] getAllowedDirections() {
        return FORWARD_DIRECTION;
    }

    @Override // com.bq.zowi.models.commands.BaseCommand, com.bq.zowi.models.commands.Command
    public void setDuration(long duration) {
        this.duration = Long.valueOf(duration);
    }

    @Override // com.bq.zowi.models.commands.MovementCommand, com.bq.zowi.models.commands.Command
    public boolean isRepeatible() {
        return false;
    }

    @Override // com.bq.zowi.models.commands.Command
    public Command copy() {
        return new ForwardBackwardCommand(getAction(), getDirection(), getDuration().longValue());
    }
}

File: zowi-core/src/main/kotlin/com/bq/zowi/models/commands/ForwardRunnerCommand.kt
package com.bq.zowi.models.commands

class ForwardRunnerCommand(
    action: Command.Action,
    direction: Command.Direction,
    duration: Long
) : MovementCommand() {

    init {
        setAction(action)
        setDirection(direction)
        setDurationRaw(duration)
    }

    override fun getAllowedActions(): Array<Command.Action> = FORWARD_ACTION
    override fun getAllowedDirections(): Array<Command.Direction> = FORWARD_DIRECTION
    override fun setDuration(duration: Long) { _duration = duration }
    override fun isRepeatable(): Boolean = false
    override fun copy(): Command = ForwardBackwardCommand(getAction(), getDirection()!!, getDuration()!!)

    companion object {
        private val FORWARD_ACTION = arrayOf(Command.Action.WALK)
        private val FORWARD_DIRECTION = arrayOf(Command.Direction.FORWARD)
    }
}
